step-api/src/main/java/org/jenkinsci/plugins/workflow/structs/DescribableHelper.java

```diff
@@ -323,14 +342,24 @@ static ParameterType of(Type type) {
                         for (Map.Entry<String,List<Class<?>>> entry : subtypesBySimpleName.entrySet()) {
                             if (entry.getValue().size() == 1) { // normal case: unambiguous via simple name
                                 try {
-                                    types.put(entry.getKey(), schemaFor(entry.getValue().get(0)));
+                                    String key = entry.getKey();
```

Comment:
I figured that it didn't matter what specific value ended up in the stack, just that the same value would show up again indicating a possible stack overflow.  Luckily they were both strings, so it made comparisons a bit easier.